internal/logger/logger.go

```diff
@@ -0,0 +1,54 @@
+package logger
+
+import (
+	"encoding/json"
+	"go.uber.org/zap"
+	"log"
+	"os"
+)
+
+var (
+	envLog  = os.Getenv("OPCAP_LOG")
+	cfg     zap.Config
+)
+
+type LogJSON struct {
+	Level         string      `json:"level"`
+	Encoding      string      `json:"encoding"`
+	OutputPaths   []string    `json:"outputPaths"`
+	EncoderConfig *EncoderCfg `json:"encoderConfig"`
+}
+
+type EncoderCfg struct {
+	MessageKey   string `json:"messageKey"`
+	LevelKey     string `json:"levelKey"`
+	LevelEncoder string `json:"levelEncoder"`
+}
+
+
+func InitLogger() *zap.Logger {
+	data := &LogJSON{
+		Level:       envLog,
```

Comment:
Instead on `envLog` can we implement this as a flag and allow the user to set envVar using the Flag itself? The Flag can be implemented at project level.

Reply:
Exactly it is being populated by an environment variable and not a flag. The idea is to implement a flag at `root.go` and it can also accept an envVar from there. Example:
```
flags.StringVar(&uploadflags.Trace, "trace", envy.Get("TRACE", "false"), "enable tracing; false by default")
```

Reply:
This way It can either be set through a flag or set through an env Var.